client/my-sites/upgrades/checkout/secure-payment-form.jsx

```diff
@@ -62,10 +64,10 @@ const SecurePaymentForm = React.createClass( {
 			return 'free-trial';
 		} else if ( this.state && this.state.userSelectedPaymentBox ) {
 			return this.state.userSelectedPaymentBox;
-		} else if ( cartValues.isCreditCardPaymentsEnabled( cart ) ) {
-			return 'credit-card';
-		} else if ( cartValues.isPayPalExpressEnabled( cart ) ) {
-			return 'paypal';
+		} else if ( cartValues.isPaymentMethodEnabled( cart, paymentMethods[ primary ] ) ) {
+			return paymentMethods[ primary ];
+		} else if ( cartValues.isPaymentMethodEnabled( cart, paymentMethods[ secondary ] ) ) {
```

Comment:
Thinking on this a bit more, maybe we could modify the selector signature so we can hide how long these preferences lists are. eg:

```
getCurrentUserPaymentMethod( state, allowedPaymentMethods )
```